src/main/java/edu/ucla/library/prl/harvester/handlers/ListJobsHandler.java

```diff
@@ -0,0 +1,48 @@
+
+package edu.ucla.library.prl.harvester.handlers;
+
+import org.apache.http.HttpStatus;
+
+import edu.ucla.library.prl.harvester.Job;
+import edu.ucla.library.prl.harvester.MediaType;
+import edu.ucla.library.prl.harvester.services.HarvestScheduleStoreService;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A handler for listing jobs.
+ */
+public final class ListJobsHandler implements Handler<RoutingContext> {
+
+    /**
+     * A proxy to the harvest schedule store service.
+     */
+    private final HarvestScheduleStoreService myHarvestScheduleStoreService;
+
+    /**
+     * @param aVertx A Vert.x instance
+     */
+    public ListJobsHandler(final Vertx aVertx) {
+        myHarvestScheduleStoreService = HarvestScheduleStoreService.createProxy(aVertx);
+    }
+
+    @Override
+    public void handle(final RoutingContext aContext) {
+        final HttpServerResponse response = aContext.response();
+
+        myHarvestScheduleStoreService.listJobs().onSuccess(jobs -> {
+            final JsonArray responseBody = new JsonArray(jobs.stream().map(Job::toJson).toList());
+
+            response.setStatusCode(HttpStatus.SC_OK)
+                    .putHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON.toString())
+                    .end(responseBody.encode());
+        }).onFailure(details -> {
+            response.setStatusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR).end(details.getMessage());
```

Comment:
In this case, I think 500 is fine for all errors, since the nano service method simply runs a database SELECT query that takes no parameters. And I think it makes sense to always consider that an internal server error from the perspective of a client.

Re: additional logging, I'm happy to add whatever you like.

Reply:
If it were me I'd just log the stacktrace in case that additional information proves useful in figuring out why it's not working. So, no value between knowing the database is down vs. it's receiving too many requests vs. bad credentials are being used, etc.?

Reply:
For all practical purposes, too many requests is not within the realm of possibility here since this is backing an admin UI driven by a small number of users. And if the app is configured with incorrect credentials, the MainVerticle will not start, because instantiation of the schedule store service will fail, because instantiating the connection pool requires correct creds.

Reply:
If the db is down, I would expect the error message to indicate some problem with the connection. So I think for all practical purposes, it covers the bases as-is.